radix-engine/src/blueprints/account/blueprint.rs

```diff
@@ -350,77 +348,104 @@ impl AccountBlueprint {
         Ok(())
     }
 
+    /// Method requires auth - if call goes through it performs the deposit with no questions asked
     pub fn deposit<Y>(bucket: Bucket, api: &mut Y) -> Result<(), RuntimeError>
     where
         Y: ClientApi<RuntimeError>,
     {
         let resource_address = bucket.resource_address(api)?;
-        let deposits_mode = Self::get_current_deposits_mode(api)?;
-
-        let is_deposit_allowed = Self::is_deposit_allowed(&deposits_mode, &resource_address, api)?;
-        if !is_deposit_allowed {
-            Runtime::assert_access_rule(rule!(require(ACCOUNT_DEPOSITS_AUTHORITY)), api)?;
-        }
-
         Self::get_vault(
             resource_address,
             |vault, api| vault.put(bucket, api),
             true,
             api,
         )?;
-
         Ok(())
     }
 
+    /// Method requires auth - if call goes through it performs the deposit with no questions asked
     pub fn deposit_batch<Y>(buckets: Vec<Bucket>, api: &mut Y) -> Result<(), RuntimeError>
     where
         Y: ClientApi<RuntimeError>,
     {
         for bucket in buckets {
             Self::deposit(bucket, api)?;
         }
-
         Ok(())
     }
 
-    pub fn safe_deposit<Y>(bucket: Bucket, api: &mut Y) -> Result<Option<Bucket>, RuntimeError>
+    /// Method is public to all - if the resource can't be deposited it is returned.
+    pub fn try_deposit<Y>(bucket: Bucket, api: &mut Y) -> Result<Option<Bucket>, RuntimeError>
```

Comment:
For consistency, `try_deposit_return_on_failure`?

Reply:
Yup, sorry did not notice with all the renames.